library/ssl_client.c

```diff
@@ -720,6 +720,30 @@ static int ssl_prepare_client_hello( mbedtls_ssl_context *ssl )
     int ret;
     size_t session_id_len;
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && \
+    defined(MBEDTLS_SSL_SESSION_TICKETS) && \
+    defined(MBEDTLS_HAVE_TIME)
+    /* Check if a tls13 ticket has been configured. */
+    if( ssl->session_negotiate->tls_version == MBEDTLS_SSL_VERSION_TLS1_3 &&
+        ssl->handshake->resume != 0 &&
+        ssl->session_negotiate != NULL &&
```

Comment:
We would gain in readability in introducing an `session_negotiate == ssl->session_negotiate` local variable as we access multiple times to this structure in this function. Otherwise I don't think we have to check that `ssl->session_negotiate != NULL` or at least we should do it as the first check here.